python/ray/serve/api.py

```diff
@@ -1712,47 +1731,79 @@ def from_yaml(cls, str_or_file: Union[str, TextIO]) -> "Application":
                 Serve YAML config files.
 
         Returns:
-            Application: a new application object containing the deployments.
+            Application: a new Application object containing the deployments.
+        """
+        return cls.from_dict(yaml.safe_load(str_or_file))
+
+    def get_ingress(self) -> Optional[Deployment]:
+        """Gets the app's ingress, if one exists.
+
+        The ingress is the single deployment with a non-None route prefix. If more
+        or less than one deployment has a route prefix, no single ingress exists,
+        so returns None.
         """
-        raise NotImplementedError()
+
+        ingress = None
+
+        for deployment in self._deployments.values():
+            if deployment.route_prefix is not None:
+                if ingress is None:
+                    ingress = deployment
+                else:
+                    return None
+
+        return ingress
 
 
 @PublicAPI(stability="alpha")
 def run(
     target: Union[DeploymentNode, Application],
+    blocking: bool = True,
     *,
     host: str = DEFAULT_HTTP_HOST,
     port: int = DEFAULT_HTTP_PORT,
-    driver: Optional[Deployment] = None,
     **kwargs,
 ) -> RayServeHandle:
     """Run a Serve application and return a ServeHandle to the ingress.
 
     Either a DeploymentNode or a pre-built application can be passed in.
     If a DeploymentNode is passed in, all of the deployments it depends on
-    will be deployed.
+    will be deployed. If there is an ingress (i.e. only one deployment with a
+    route prefix), its handle will be returned.
```

Comment:
don't mention that we use route prefix to determine it, that's an implementation detail